csrc/multidevice/utils.cpp

```diff
@@ -633,13 +633,18 @@ bool isInnerResharding(Expr* expr) {
   return false;
 }
 
-void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs) {
+void shardAllLike(
+    TensorView* ref,
+    std::vector<TensorView*> tvs,
```

Comment:
```suggestion
    const std::vector<TensorView*>& tvs,
```

Reply:
In general, `T` leads to more or as many copies than `const T&` unless all the callers are very careful with `std::move`.